### PR TITLE
stop testing Nim pre-1.6 in CI

### PR DESCRIPTION
**Pull request by tersec**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -166,12 +166,6 @@ jobs:
...
             echo -e "\nTesting with '--gc:orc':\n"
-            if env TEST_LANG="c" NIMFLAGS="${NIMFLAGS} --gc:orc" nimble test; then
-              echo "Nim devel with --gc:orc works again! Please remove this check in ci.yml"
-              false
-            fi
-            if env TEST_LANG="cpp" NIMFLAGS="${NIMFLAGS} --gc:orc" nimble test; then
-              echo "Nim devel with --gc:orc works again! Please remove this check in ci.yml"
-              false
-            fi
+            env TEST_LANG="c" NIMFLAGS="${NIMFLAGS} --gc:orc" nimble test
```
Comment (arnetheduck) [inline .github/workflows/ci.yml:169]: orc is default in devel, it would be more relevant to test devel-refc separately

Comment (arnetheduck) [reply .github/workflows/ci.yml:169]: oh, we do that above - well, I'd remove --gc:orc regardless to catch "defaults" whatever they might be

Comment (tersec) [reply .github/workflows/ci.yml:169]: The .nimble file also handles this: https://github.com/status-im/nim-ssz-serialization/blob/master/ssz_serialization.nimble

It should go in one place or another, but don't have strong view on where.

But I'm not sure what you mean by "test devel-refc separately": the idea is that as long as Status hasn't transitioned away from refc, anyversion-refc will be relevant. So both ORC and refc should be tested, post-1.6. Neither is any more "separate" than the other.

Comment (tersec) [reply .github/workflows/ci.yml:169]: Sure, one can also have implied-default-model and refc separately, post-1.6.

If Status also starts depending on ORC, then that then needs to be added specifically, regardless of defaults, but it doesn't yet.

Comment (tersec) [reply .github/workflows/ci.yml:169]: Oh, it doesn't actually work in ORC in `devel` (but apparently does in 2.0):
```
[Suite] Merkle proofs
  [OK] get_branch_indices
  [OK] get_path_indices
  [OK] get_helper_indices
  [OK] verify_merkle_multiproof
Traceback (most recent call last)
/home/runner/.nimble/pkgs/unittest2-0.0.6/unittest2.nim(757) test_ssz_optional
/home/runner/.nimble/pkgs/unittest2-0.0.6/unittest2.nim(682) suiteEnded
/home/runner/.nimble/pkgs/unittest2-0.0.6/unittest2.nim(309) suiteEnded
/home/runner/work/nim-ssz-serialization/nim-ssz-serialization/nim/lib/system/arc.nim(211) isObjDisplayCheck
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
  [OK] is_valid_merkle_branch
```
So not sure why the initial runs triggered the "re-enable ORC CI" message, which is/was supposed to run anyway only when ORC actually runs.

Fixing whatever's happening there is orthogonal to the gist of this PR.

**Comment by arnetheduck** [comment]
I do wonder if we should start expressing version constraints more specifically in nimble files for nim - ie we have random versions sprinkled all over the place, but when we discover that a particular library is (significantly) broken with a particular Nim version, the correct thing to do would be to write that down as a version constraint in its nimble file, thus communicating it to users in general

**Comment by tersec** [comment]
For the record, a couple of repros for the CI issue here; it seems `faststreams/outputs` related.

[repro.txt](https://github.com/status-im/nim-ssz-serialization/files/11327331/repro.txt)
[repro2.txt](https://github.com/status-im/nim-ssz-serialization/files/11327332/repro2.txt)

**Comment by tersec** [comment]
> I do wonder if we should start expressing version constraints more specifically in nimble files for nim - ie we have random versions sprinkled all over the place, but when we discover that a particular library is (significantly) broken with a particular Nim version, the correct thing to do would be to write that down as a version constraint in its nimble file, thus communicating it to users in general

https://github.com/status-im/nim-ssz-serialization/pull/45/commits/ecf42b0a9b6d351e3f6287059d580743ed9640e5 seems to hit all the current `devel`/`version-2-0` conditions.

In general, I don't disagree, but here, it's specifically `devel`, not a released version, which it's incompatible with (in ORC specifically). It's reasonable enough to count incompatible-with-default-memory-manager as incompatible, but `devel` is not a targetable version as such, so I don't think this concern applies here.

It does, somehow, work under `version-2-0`, so despite preferring to work within the `.nimble` file, this adds the CI-side workaround, because from the `.nimble` perspective, and I just checked a couple hours ago, both `devel` and `version-2-0` are Nim version 1.9.3.

Once either Nim 2.0 acquires its final `(NimMajor, NimMinor)` tuple values or this `devel` + `ORC` + 64-bit target (and no other combination I've found) bug is fixed, then one can remove the `.github/workflows/ci.yml` workaround.

**Comment by arnetheduck** [comment]
> it's specifically devel,

indeed, supporting devel remains best-effort - the version tracking would note only released nim versions.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (etan-status) [reply .github/workflows/ci.yml]: The explicit `--gc:orc` is needed for the 1.6 branch, it is only default from 2.0 onwards, no?

Comment (etan-status) [reply .github/workflows/ci.yml]: Ah, `if [[ "${{ matrix.branch }}" == "devel" ]]; then`, got it

Comment (etan-status) [reply .github/workflows/ci.yml]: About "The .nimble file also handles this", I think Nim regularly builds our projects using the nimble file, so keeping it erroring there makes them aware of the issue if they try to make our projects build, while putting it in our custom CI allows us to still have a green CI with known errors suppressed for ourselves.

**Review by etan-status — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tersec) [reply .github/workflows/ci.yml]: That's the status quo and also the situation this PR would perpetuate that
https://github.com/status-im/nim-ssz-serialization/blob/3f8946ab2de304a9130aa80874bc2e4025d62303/ssz_serialization.nimble#L28-L33

This PR doesn't change that, and it accomplishes what you suggest, and which I agree with.

**Comment by tersec** [comment]
Maybe it's https://github.com/status-im/nim-faststreams/issues/39 ?

**Pull request merged**